### PR TITLE
4 get article by id

### PR DESCRIPTION
**Pull request by VenturingForth**
Created a filepath for accessing articles by article_id.

Added 400 error handling for invalid IDs.

Added 404 error handling for valid but non-existant IDs.

Updated endpoints.json documentation.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -43,4 +43,40 @@ describe("GET /api", () => {
...
+    describe("/articles", () => {
+        describe("/:article_id", () => {
+            test("200: Should retrieve an article by id", () => {
+                return request(app)
+                .get('/api/articles/1')
+                .expect(200)
+                .then((res) => {
+                    expect(res.body.article).toHaveProperty("author", expect.any(String));
+                    expect(res.body.article).toHaveProperty("title", expect.any(String));
+                    expect(res.body.article).toHaveProperty("article_id", expect.any(Number));
```
Comment (pippaamy) [inline __tests__/app.test.js:55]: we can expect this article_id to be 1 here :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -43,4 +43,40 @@ describe("GET /api", () => {
             })
         })
     })
+    describe("/articles", () => {
+        describe("/:article_id", () => {
+            test("200: Should retrieve an article by id", () => {
+                return request(app)
+                .get('/api/articles/1')
+                .expect(200)
+                .then((res) => {
```
Comment (pippaamy) [inline __tests__/app.test.js:52]: you could destructure {body} here to save some time

File: __tests__/app.test.js [hunk 3]
```diff
@@ -43,4 +43,40 @@ describe("GET /api", () => {
...
+                    expect(res.body.article).toHaveProperty("created_at", expect.any(String));
+                    expect(res.body.article).toHaveProperty("votes", expect.any(Number));
+                    expect(res.body.article).toHaveProperty("article_img_url", expect.any(String));
+                    console.log(res.body);
+                })
+            })
+            test("404: Should return \"Article ID not found\" if given valid but non-existent ID", () => {
+                return request(app)
+                .get('/api/articles/999')
+                .expect(404)
```
Comment (pippaamy) [inline __tests__/app.test.js:67]: great error testing

File: controllers/articles-controllers.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const { fetchArticleById } = require("../models/articles-models.js")
+
+module.exports.getArticleById = (req, res, next) => {
+    const { article_id } = req.params;
+    return fetchArticleById(article_id).then((article) => {
+        res.status(200).send({article: article})
```
Comment (pippaamy) [inline controllers/articles-controllers.js:6]: this can be sent back as just {article} if you want

**Review by pippaamy — COMMENTED** [review]
Great job, popped some comments to have a think about :)

File: __tests__/app.test.js (hunk 1 above)
Comment (VenturingForth) [reply __tests__/app.test.js:55]: This has been refactored.

File: __tests__/app.test.js (hunk 2 above)
Comment (VenturingForth) [reply __tests__/app.test.js:52]: This has been refactored.

File: __tests__/app.test.js (hunk 3 above)
Comment (VenturingForth) [reply __tests__/app.test.js:67]: Thanks!

File: controllers/articles-controllers.js (hunk 4 above)
Comment (VenturingForth) [reply controllers/articles-controllers.js:6]: This has been refactored.

**Pull request merged**